File: src/main/standard_problems/print_binary.py
import sys
def printBin(arr,n):
    if n < 0:
        print (arr)
    else :
        arr[n] = 0
        printBin(arr,n-1 )
        arr[n] = 1
        printBin(arr,n-1 )
def printBinary(n):
    array = [None] * n
    printBin(array,n-1)


printBinary(int(sys.argv[1]))

File: src/main/java/recursionandbacktrack/PrintBinary.java
package recursionandbacktrack;

public class PrintBinary {
  public static void printBinary(int n) {
    printBinary(n,"");
  }

  private static void printBinary(int n , String s) {
    if (n == 0) {
      System.out.println(s);
      return;
    }
    printBinary(n-1,s + "0");
    printBinary(n-1,s + "1");
  }


  public static void main(String[] args) {
    printBinary(3);
  }
}
